config/config.go

```diff
@@ -1,18 +1,275 @@
+var (
+	Debug          bool
+	Dryrun         bool
+	VersionFlagSet bool
+	// ConfigFilePath is `.` if not set through the `-configFile` flag
+	ConfigFilePath string
+
+	// ProcWorkingDir is set through calling os.Getwd()
+	ProcWorkingDir string
+	ExePath        string
+)
+
+// Data represents the config data used in the day-to-day running of can
+//
+//	TODO this may be vague in definition for the sake of its legibility in use
+//	TODO check for redundancy
+type Data struct {
+	Generator
+	Template
+
+	// left public due to it's need to be unmarshalled by Data.Load()
+	TemplatesDir string `yaml:"templatesDir"`
+
+	// OpenAPIFile represents the path to the yaml OpenAPI 3 file to render
+	OpenAPIFile    string
+	absOpenAPIPath string
+
+	OutputPath    string
+	absOutputPath string
+}
+
+type Generator struct {
+	ModuleName string
+
+	// BasePackageName represents the
+	BasePackageName string
+}
+
+// Template is populated based on it's Name variable set as a CLI flag
+type Template struct {
+	Name string
+
+	// Directory should be ./templates/${Name} by default
+	Directory    string
+	absDirectory string
+}
+
+func (d *Data) Load() (err error) {
+	// Setup config pre-unmarshalling. This assumes we don't change directory before this is executed
+	ProcWorkingDir, err = os.Getwd()
+	if err != nil {
+		return fmt.Errorf("Config.load:: could not determine working directory: %w\n", err)
+	}
+	ExePath, err = os.Executable()
+	if err != nil {
+		return fmt.Errorf("Config.load:: could not determine executable path: %w\n", err)
+	}
+
+	// flags
+	if !flag.Parsed() { // Sorts out buggy tests
+		flag.BoolVar(&VersionFlagSet, "version", false, "Print Can version and exit")
+		flag.BoolVar(&Debug, "debug", false, "Enable debug logging")
+		flag.StringVar(&ConfigFilePath, "configFile", ".", "Specify which config file to use")
+		flag.BoolVar(&Dryrun, "dryrun", false,
+			"Toggles whether to perform a render without writing to disk."+
+				"This works particularly well in combination with -debug")
+		flag.StringVar(&d.Template.Name, "template", "", "Specify which template set to use")
+		flag.Parse()
+	}
+
+	absCfgPath, err := filepath.Abs(ConfigFilePath)
+	if err != nil {
+		return fmt.Errorf("could not resolve relative config path: %w", err)
+	}
+	ConfigFilePath = absCfgPath
+
+	if VersionFlagSet {
+		fmt.Printf("Can Version: %s\n", SemVer)
+		os.Exit(0)
+	}
+
+	// config load
+	if Debug {
+		fmt.Printf("[v%s]::Using config file \"%s\".\n", SemVer, ConfigFilePath)
+	}
+	viper.SetConfigFile(ConfigFilePath)
+
+	err = viper.ReadInConfig()
+	if err != nil {
+		return fmt.Errorf("loadConfig:: could not read config file: %w\n", err)
+	}
+
+	err = viper.Unmarshal(&d)
+	if err != nil {
+		return fmt.Errorf("loadConfig:: could not parse config file: %w\n", err)
+	}
+
+	// This should always happen at the end of this function
+	// Handle Templates
+	if d.Template.Name == "" {
+		fmt.Printf("template is a required flag\nexiting...")
```

Comment:
why not allow reading the template from the config file?

Reply:
released this restriction, as discussed on discord